client/components/AutocompleteInput/components/PackageNameElement.tsx

```diff
@@ -0,0 +1,7 @@
+interface Props extends React.HTMLAttributes<HTMLElement> {
+  as?: 'h1'
+}
+
+export function PackageNameElement({ as, ...props }: Props) {
+  return as === 'h1' ? <h1 {...props} /> : <span {...props} />
+}
```

Comment:
nit / optional:  I'm not a big fan of splitting components into new files for the sake of it. If a component is small enough, and will only ever be used in one component, keeping it colocated avoids unnecessary context switching.

Reply:
✅